client/src/components/Toasts/Toast.test.js

```diff
@@ -67,23 +70,69 @@ describe('Notification Toast', () => {
       })
     );
     expect(queryByRole('alert')).toBeDefined();
-    expect(queryByRole('alert')).toHaveTextContent(/RStudio-S...cvserver is now running/);
-    expect(queryByRole('alert')).not.toHaveTextContent(/RStudio-S...cvserver finished successfully/);
+    expect(queryByRole('alert')).toHaveTextContent(
+      /RStudio-S...cvserver is now running/
+    );
+    expect(queryByRole('alert')).not.toHaveTextContent(
+      /RStudio-S...cvserver finished successfully/
+    );
+  });
+});
+
+describe('Toast Message', () => {
+  it('shows data file error toast message', () => {
+    const { getByText } = renderComponent(
+      <ToastMessage notification={dataFilesError} />,
+      mockStore({
+        systems: initialSystemState,
+        projects: {
+          listing: {
+            projects: []
+          }
+        }
+      })
+    );
+    expect(getByText('Move failed')).toBeDefined();
+  });
+  it('shows upload message including system information', () => {
+    const { getByText } = renderComponent(
+      <ToastMessage notification={dataFilesUpload} />,
+      mockStore({
+        systems: systemsFixture,
+        projects: projectsFixture
+      })
+    );
+    expect(getByText('File uploaded to My Data (Frontera)/')).toBeDefined();
+  });
+  it('shows upload message including system information for shared workspace', () => {
+    const { getByText } = renderComponent(
+      <ToastMessage notification={dataFilesUploadToSharedWorkSpace} />,
+      mockStore({
+        systems: systemsFixture,
+        projects: projectsFixture
+      })
+    );
+    expect(getByText('File uploaded to Test Project Title/')).toBeDefined();
   });
 });
 
 describe('getToastMessage', () => {
   it('returns expected job response', () => {
-    expect(getToastMessage(jobStatusUpdatePending)).toEqual('RStudio-S...cvserver is processing');
+    expect(getToastMessage(jobStatusUpdatePending)).toEqual(
+      'RStudio-S...cvserver is processing'
```

Comment:
Do we have a truncation strategy in code? If so, might want to import and run that function here instead just in case truncation strategy changes.

Reply:
>  just in case truncation strategy changes.
👍  on the idea.     Just checked and there is a `truncateMiddle` but it is only used on the first part of the message.  Adding now but if you or other reviewers feel like it complicates things, I can revert.

Reply:
@jchuahtacc , actually, i'm gonna revert as we have to add such implementation details in the test (like the `20`) so it kinda degrades the initial goal of your comment of keeping things generic enough to handle truncation strategy changes.